src/rust/rust-proto/src/plugin_work_queue.rs

```diff
@@ -177,44 +249,52 @@ impl From<GetExecuteGeneratorRequest> for GetExecuteGeneratorRequestProto {
 }
 
 pub struct GetExecuteGeneratorResponse {
-    execution_job: ExecutionJob,
-    request_id: uuid::Uuid,
+    pub execution_job: Option<ExecutionJob>,
+    pub request_id: i64,
 }
 
 impl TryFrom<GetExecuteGeneratorResponseProto> for GetExecuteGeneratorResponse {
     type Error = PluginWorkQueueDeserializationError;
 
     fn try_from(value: GetExecuteGeneratorResponseProto) -> Result<Self, Self::Error> {
-        let request_id = value
-            .request_id
-            .ok_or(Self::Error::MissingRequiredField(
-                "GetExecuteGeneratorResponse.request_id",
-            ))?
-            .into();
-        let execution_job = value
-            .execution_job
+        let request_id = value.request_id;
+        let maybe_job = value.maybe_job.ok_or(Self::Error::MissingRequiredField(
+            "GetExecuteGeneratorResponseProto.maybe_job",
+        ))?;
+        let execution_job = match maybe_job {
+            get_execute_generator_response::MaybeJob::Job(job) => Some(job),
+            get_execute_generator_response::MaybeJob::NoJobs(_) => None,
+        };
```

Comment:
Can this `MaybeJob` -> `Option` (and back) conversion be implemented explicitly with `From`/`Into`?

I generally prefer this kind of behavior to be expressed in terms of traits, rather than buried away in individual method bodies, for clarity, explicitness, reuse, documentation, testability, etc.

Reply:
Possibly, I'm not sure if orphan rules allow us to implement traits we don't own on types like Option<T>, even if we own T. At one point it was forbidden, I don't know if they've changed that - though I know there has been discussion for years.

Reply:
Looks like orphan rules now allow this, neat. Will implement.

Reply:
:tada: